cmake/compiler_setup.cmake

```diff
@@ -116,9 +108,20 @@ if(MSVC)
 
     # unary minus operator applied to unsigned type, result still unsigned
     /wd4146
+
+    # character cannot be represented in the current code page
+    /wd4566
```

Comment:
Question: why is this change necessary?

Reply:
MSVC emits a warning that one of the unicode literals in one of our tests cannot be represented in the current code page (in my case Windows-1252). This is irrelevant to the test and is generally irrelevant, so I've suppressed the warning.